src/main/java/net/fabricmc/loader/discovery/ModResolver.java

```diff
@@ -85,6 +86,22 @@ private static IVecInt toVecInt(IntStream stream) {
 		return new VecInt(stream.toArray());
 	}
 
+	public static String getReadablePath(FabricLoader loader, ModCandidate c) {
```

Comment:
Is there a reason why this requires the loader instance? Considering that it's currently a singleton (and always assigned before it's used), and that most of the codebase *doesn't* pass the loader around, and this only uses the public API of FabricLoader, might it make sense to drop this argument?

(Sidenote: I'm mostly bringing this up as I'm copy-pasting this into my other PRs so as to avoid providing duplicate functionality, and to make merging them much simpler).

Reply:
I'd honestly like to change everything where possible to just pass a loader instance around.

Reason being I'd like to abstract the actual creation of the loader instance a bit so other launch mechanisms could use it.

Reply:
I'd actually like to see passing a loader instance around too. But the reason here for passing the FabricLoader instance is way more simple: this class already imports the FabricLoader impl class (or relies on it otherwise), so I had two options: either use the deprecated FabricLoader.INSTANCE field, or use the fully qualified name for the api method FabricLoader.getInstance(). Both options kinda sucked imo so I just passed it as an argument. Not opposed to changing this though.

Reply:
Fair enough!